src/FEValues/CellMultiValues.jl

```diff
@@ -0,0 +1,288 @@
+end
+
+getnbasefunctions(funvals::FunctionValues) = size(funvals.N, 1)
+@propagate_inbounds shape_value(funvals::FunctionValues, q_point::Int, base_func::Int) = funvals.N[base_func, q_point]
+@propagate_inbounds shape_gradient(funvals::FunctionValues, q_point::Int, base_func::Int) = funvals.dNdx[base_func, q_point]
+@propagate_inbounds shape_symmetric_gradient(funvals::FunctionVectorValues, q_point::Int, base_func::Int) = symmetric(shape_gradient(funvals, q_point, base_func))
+@propagate_inbounds shape_divergence(funvals::FunctionScalarValues, q_point::Int, base_func::Int) = sum(funvals.dNdx[base_func, q_point])
+@propagate_inbounds shape_divergence(funvals::FunctionVectorValues, q_point::Int, base_func::Int) = tr(funvals.dNdx[base_func, q_point])
+@propagate_inbounds shape_curl(funvals::FunctionVectorValues, q_point, base_func) = curl_from_gradient(shape_gradient(funvals, q_point, base_func))
+
+
+struct CellMultiValues{dim,T,RefShape,FVS<:NamedTuple} <: CellValues{dim,T,RefShape}
+    geo_values::GeometryValues{dim,T,RefShape}
+    fun_values::FVS
+    detJdV::Vector{T}
+    qr::QuadratureRule{dim,RefShape,T}
+end
+function CellMultiValues(;cvs...)
+    # cvs::Pairs{Symbol, CellValues, Tuple, NamedTuple}, cf. foo(;kwargs...) = kwargs
+    #@assert allequal(typeof(cv.qr) for (_, cv) in cvs)
+    #@assert allequal(length(getweights(cv.qr)) for (_, cv) in cvs)
+    cv1 = first(values(cvs))
+    @assert all(==(typeof(cv1.qr)), typeof(cv.qr) for (_, cv) in cvs)
+    @assert all(==(length(getweights(cv1.qr))), length(getweights(cv.qr)) for (_, cv) in cvs)
+    
+    geo_values = GeometryValues(cv1)
+    fun_values = NamedTuple(key=>create_function_values(cv) for (key, cv) in cvs)
+    return CellMultiValues(geo_values, fun_values, cv1.detJdV, cv1.qr)
+end
+
+# Quadrature
+# getnquadpoints(::QuadratureRule) would be nice...
+getnquadpoints(cv::CellMultiValues) = length(cv.qr.weights)
+
+# Geometric functions
+getngeobasefunctions(cv::CellMultiValues) = getngeobasefunctions(cv.geo_values)
+getdetJdV(cv::CellMultiValues, q_point::Int) = cv.detJdV[q_point]
+geometric_value(cv::CellMultiValues, q_point::Int, base_func::Int) = geometric_value(cv.geo_values, q_point, base_func)
+
+# FunctionValues functions: call like with CellValues, but foo(cv[:name], args...)
+Base.getindex(cmv::CellMultiValues, key::Symbol) = getindex(cmv.fun_values, key)
+
+# Standard identity mapping (scalar function / lagrange vector)
+function apply_mapping!(funvals::FunctionValues{dim}, q_point::Int, Jinv::Tensor{2,dim}) where dim
+    @inbounds for j in 1:getnbasefunctions(funvals)
+        funvals.dNdx[j, q_point] = funvals.dNdξ[j, q_point] ⋅ Jinv
+    end
+    return nothing
+end
+
+#= TODO: This function could be generalized as follows, but does not make sense before we 
+   implement other mappings (i.e. Piola-mappings)
+"""
+    apply_mapping!(funvals::FunctionValues, q_point::Int, detJ, J, Jinv, geo_values)
```

Comment:
Did you have something in mind for how to dispatch on this? I thought the other day that we should change the parametrization of https://github.com/Ferrite-FEM/Ferrite.jl/blob/247d7c3efdd6325a78c7274180d4a439921dcc78/src/FEValues/cell_values.jl#L89 to instead be parametrized on the function and geometric interpolation
```julia
struct CellScalarValues{IPF, IPG}
```
to make this kind of dispatches possible.

Reply:
Yes, that is the primary solution I've thought about.

Reply:
Maybe we should gather somewhere the ideas regarding breaking changes of the FEValues to fix some open issues, so we can group the PRs and do not run into the issue of having several breaking changes over several releases.

Reply:
That would indeed be good, and if those are getting ready would be nice to include in the 0.4 release unless it causes a too long hold-back (I'm eager to start adapting the the merged dofhandlers:)) 
(This PR is actually non-breaking, though)